### ISSUE TITLE
Change the default time evolution method

### ISSUE THREAD

**Issue by liwt31** [shuaigroup/Renormalizer#143]
Currently, the default time evolution method is `prop_and_compress`, which is not very popular. Maybe it's time to change it to TDVP_PS or TDVP_VMF?

**Comment by jjren** [comment]
RK4 is actually more black box than TDVP_PS and TDVP_VMF to get not bad results (of course not the best) for most Hamiltonians.
Because TDVP methods need a good/correct initialization (renormalized space expansion) to avoid the evolution stuck in a wrong subspace. It depends on if the user understands and runs the algorithm correctly.
For example, in the transport problem, all the one-exciton state should be included in the initial renormalized space carefully.
If not, you will get an artificial localization solution.
I don't know whether 2-site TDVP_PS can avoid this kind of problem. But RK4 is more robust, at least get a "correct" answer.
Do you think so?

**Comment by liwt31** [comment]
Makes sense. However I'd like to add that the step control in TDVP_VMF is a unique advantage.

Then it again boils down to the documentation. We should list all the possible schemes and their (dis)advantages. Some of them are published with well-supported data (like the 2020JCP work). Some of them are more empirical.

**Comment by jjren** [comment]
I totally agree with you about improving the documentation for all the evolution algorithms. 
It is better if you can write a template and set a doc structure, then I can add content.

For the current situation, I think we can switch the default algorithm to P&C TD RK (general explicit RK propagator for both time-dependent/independent H), with which the time-step can be controlled adaptively (e.g Cash-Karp45, RKF45).

**Comment by liwt31** [comment]
> I don't know whether 2-site TDVP_PS can avoid this kind of problem. 

I think not if the electronic sites are separated by phonon sites. Maybe this is worth documenting too.

> It is better if you can write a template and set a doc structure, then I can add content.

No problem

> we can switch the default algorithm to P&C TD RK

OK. Could you do it while adding the documentation? I assume you're more familiar with the setup. Honestly I rarely use this method and I'm under the impression that the step-size control for P&C RKF is not particularly reliable. Sometimes the step size simply keeps shrinking to infinitely small in one step.

**Comment by jjren** [comment]
> OK. Could you do it while adding the documentation? I assume you're more familiar with the setup. Honestly I rarely use this method and I'm under the impression that the step-size control for P&C RKF is not particularly reliable. Sometimes the step size simply keeps shrinking to infinitely small in one step.

That's true in some cases because the error due to time-step is probably overwhelmed by the truncation error in one single step. In this case, weird results will occur. 
In my experience, one simple prescription is that RK45 should be used in combination with adaptive bond dimension ($\eta_bond dimension cutoff < \eta_{time step error}$).
I will switch the current scheme to this evolution scheme, make some tests, and add documentation.
